docs/guide/best-practices.md

```diff
@@ -4,6 +4,156 @@ outline: deep
 
 # Best practices
 
+## Standalone components
+
+Since Angular 15 [standalone components](https://angular.io/guide/standalone-components) are now officially supported. They offer many advantages and should be used as much as possible, both for new developments and when refactoring existing code.
+
+Standalone components have the following differences with legacy "non-standalone" ones:
+
+- Standalone components are _not_ declared in Angular Modules; instead, they act as their own module, declaring their own dependencies and providers
+- Standalone components can import either Angular modules or other standalone components
+
+## Testing
+
+### Unit tests
+
+#### Mocking services
+
+One of the advantages of the Angular [dependency injection](https://angular.io/guide/dependency-injection-overview) system is the ability to use mock classes when testing a component or services.
+
+This can be done by specifying providers in `TestBed.configureTestingModule`:
+
+```ts
+import { MyService } from './my-service'
+import { MyComponent } from './my-component.component'
+// ...
+
+class MyServiceMock {
+  someData$ = new BehaviorSubject([])
+}
+
+describe('MyComponent', () => {
+  let component: MyComponent
+  let myService: MyService
+  let fixture: ComponentFixture<MyComponent>
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: MyService,
+          useClass: MyServiceMock,
+        },
+        // ....
+      ],
+    })
+
+    fixture = TestBed.createComponent(MyComponent)
+    myService = TestBed.inject(MyService)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+  }).compileComponents()
+
+// ...
+```
+
+::: tip
+When defining providers, **`useClass` should be preferred to `useValue`** to avoid side effects between test cases.  
+Indeed, `useClass` means a new instance of a dependency is created for each test, whereas `useValue` reuses the same instance across all tests.
+:::
+
+#### Targeting nodes in the DOM
+
+When testing components, it is often useful to target nodes in the resulting DOM to check their presence/absence, text content etc.
+
+To do this, it is recommended to add a `data-test` attribute to the target node in the component's HTML template, for instance:
+
+```html
+<div>My component</div>
+<div *ngIf="data$ | async as currentData" data-test="show-data">
+  {{ currentData }}
+</div>
+```
+
+This makes the HTML template easier to read and avoid confusion with e.g. CSS classes.
+
+#### Mocking translations
```

Comment:
Is this valid for Storybook too?

Reply:
Hmm, good question; I have the impression that translations just don't work in Storybook, which is often not very visible because UI components tend to have their labels provided to them already translated. We should make Storybook load the default EN translations I guess.

Reply:
I took a look and actually translations work just fine in Storybook, I will document how to enable them.